core/src/transport/mod.rs

```diff
@@ -47,73 +46,85 @@ pub use self::choice::OrTransport;
+/// whose connection setup incorporates all earlier upgrades followed by the new upgrade,
+/// i.e. the order of the upgrades is significant.
 ///
 /// > **Note**: The methods of this trait use `self` and not `&self` or `&mut self`. In other
 /// >           words, listening or dialing consumes the transport object. This has been designed
 /// >           so that you would implement this trait on `&Foo` or `&mut Foo` instead of directly
 /// >           on `Foo`.
 pub trait Transport {
-    /// The raw connection to a peer.
+    /// A data stream (i.e. connection) created by the transport.
     type Output;
-    /// Error that can happen when dialing or listening.
+
+    /// An error that occurred during connection setup.
     type Error: error::Error;
 
-    /// The listener produces incoming connections.
+    /// A stream of incoming connections.
     ///
     /// An item should be produced whenever a connection is received at the lowest level of the
-    /// transport stack. The item is a `Future` that is signalled once some pre-processing has
-    /// taken place, and that connection has been upgraded to the wanted protocols.
+    /// transport stack. The item is a [`Future`] that resolves to a connection once all
+    /// protocol upgrades have been applied.
     type Listener: Stream<Item = (Self::ListenerUpgrade, Multiaddr), Error = Self::Error>;
 
-    /// After a connection has been received, we may need to do some asynchronous pre-processing
-    /// on it (e.g. an intermediary protocol negotiation). While this pre-processing takes place,
-    /// we want to be able to continue polling on the listener.
+    /// A pending, inbound connection, obtained from the stream of incoming connections.
+    ///
+    /// After a connection has been accepted by the transport, it may need to go through
+    /// asynchronous post-processing (i.e. protocol upgrade negotiations). Such
+    /// post-processing should not block the `Listener` from producing the next
+    /// connection, hence further connection setup proceeds asynchronously.
+    /// One a `ListenerUpgrade` future resolves, the connection setup is complete.
     type ListenerUpgrade: Future<Item = Self::Output, Error = Self::Error>;
 
-    /// A future which indicates that we are currently dialing to a peer.
+    /// A pending, outbound connection, obtained through dialing on the transport.
     type Dial: Future<Item = Self::Output, Error = Self::Error>;
 
-    /// Listen on the given multiaddr. Returns a stream of incoming connections, plus a modified
-    /// version of the `Multiaddr`. This new `Multiaddr` is the one that that should be advertised
-    /// to other nodes, instead of the one passed as parameter.
+    /// Listens on the given [`Multiaddr`], producing a stream of pending, inbound connections.
     ///
-    /// > **Note**: The reason why we need to change the `Multiaddr` on success is to handle
```

Comment:
This style of putting `> ` in front of notes is what the documentation of the standard library does, so I followed it.

Reply:
Ok.